torch/distributed/pipeline/sync/checkpoint.py

```diff
@@ -302,32 +312,49 @@ def forward(
         rng_states: Deque[RNGStates],
         function: Function,
         input_atomic: bool,
-        *input: Tensor,
+        *inputs,
     ) -> Tensor:
         ctx.recomputed = recomputed
         ctx.rng_states = rng_states
 
         ctx.function = function
         ctx.input_atomic = input_atomic
-        ctx.save_for_backward(*input)
+        ctx.inputs = inputs
+        if input_atomic:
+            tensors = [inputs[0]]
+        else:
+            tensors = []
+            for input in enumerate(inputs):
+                if torch.is_tensor(input):
```

Comment:
doesn't enumerate always return a tuple? So this is always false, or am I missing something?